src/components/Characters/Characters.js

```diff
@@ -1,54 +1,147 @@
-import React, {useState, useEffect} from 'react'
-import Character from '../Characters/Character.js'
-import './Characters.css'
-
-
-const Characters = () => {
-    const [url, setUrl] = useState('https://rickandmortyapi.com/api/character/?name=')    // Fetching charact. API
-    const [info, setInfo] = useState({})                  
-    const [results, setResults] =useState([])
-    const [search, setSearch] =useState('')
-  
-    useEffect(() => {                   //Checking some data objects (testing purposes) 
-      console.log('url: ', url)
-      console.log('info: ', info)
-      console.log('results', results)
-      console.log('search: ', search)
-    }, [info, url,search, results])
-  
-    useEffect(()=>{                     //combined url + result in input for search filter purpose
-        fetch(`${url}${search}`, {
-          method: 'GET'
-        }).then((response) => {
-          return response.json()
-        }).then((result) => {
-            setInfo(result.info)
-            setResults(result.results)
-        }).catch((error) => {
-            console.log(error)
-        })
-    }, [search])                        //Serach as a main dependency for this 
-  
-    return (
-        <div className="all-wrapper">
-            <div>
-                <input onChange={(e) => {
-                setSearch(e.target.value)
+import React, { useState, useEffect } from 'react';
+import Character from './Character';
+import './Characters.css';
+import FilterIcon from '../../content/icons/filter.png';
+import ArrowLeft from '../../content/icons/arrow-left.png';
+import ArrowRight from '../../content/icons/arrow-right.png';
+import Facebook from '../../content/icons/facebook.png';
+import LinkedIn from '../../content/icons/linkedin.png';
+import Github from '../../content/icons/github.png';
+
+import Filter from '../Filter/Filter';
+
+function Characters() {
+  const [url, setUrl] = useState('https://rickandmortyapi.com/api/character/'); // Fetching charact. API
+  const [info, setInfo] = useState({});
+  const [results, setResults] = useState([]);
+  const [search, setSearch] = useState('');
+  const [page, setPage] = useState(1);
+
+  const [filter, setFilter] = useState(false);
+
+  const [error, setError] = useState(false);
+  const [addition, setAddition] = useState('');
+
+  const getDataValue = (value) => {
+    setAddition(value);
+  };
+
+  const getErrorValue = (value) => {
+    setError(value);
+  };
+
+  const getSearchValue = (value) => {
+    setSearch(value);
+  };
+
+  useEffect(() => { // Checking some data objects (testing purposes)
+    console.log('url: ', url);
+    console.log('info: ', info);
+    console.log('results', results);
+    console.log('search: ', search);
+  }, [info, url, search, results]);
+
+  useEffect(() => {
+    const controller = new AbortController(); // combined url + result in input for
```

Comment:
make sure it works, use slow 3G simulation in Chrome dev tools

Reply:
I checked 2 versions. One with an Abort controller and one without. So performance is the same as well as lack of errors. There are no errors in both cases.